### ISSUE TITLE
user module doesn't work if specifying a password in the module

### ISSUE THREAD

**Issue by cunlifs** [IBM/ansible-power-aix#173]
Hi, using the latest version of the collection, creating a brand new user with the user module fails if specifying the password in the module.

For example this works:
  vars:
    user_name: "user2"
    group_list: "system"
    uid: "1012"

  tasks:
    # Add/Update users
    - name: Add/Update user {{ user_name }}
      ibm.power_aix.user:
        name: "{{ user_name }}"
        state: present
        attributes: 
          id: "{{ uid }}" 
          groups: "{{ group_list }}"

and creates the new user OK. 

However however we try to define the password it fails:
  vars:
    user_name: "user2"
    group_list: "system"
    uid: "1012"
    password: "abc1234"

  tasks:
    # Add/Update users
    - name: Add/Update user {{ user_name }}
      ibm.power_aix.user:
        name: "{{ user_name }}"
        state: present
        password: "{{ password }}"
        attributes: 
          id: "{{ uid }}" 
          groups: "{{ group_list }}"

Throws up this error:
fatal: [lab-aix-1]: FAILED! => changed=false 
  msg: |2-
    Failed to modify attributes for the user: user2
  rc: 17
  stderr: |-
    3004-692 Error changing "id" to "1012" : Account exists.
  stderr_lines: <omitted>
  stdout: ''
  stdout_lines: <omitted>

The user was completely removed befor re-adding, so why is it saying 'failed to modify attributes for user'?

Ansible version: 2.9.17
AIX level of client: 7200-03-03-1914

**Comment by d-little** [comment]
You will need to switch the `state` from `present` to `modify` for existing users.

I do believe there's a feature request in this issue:  it shouldn't matter if the user has already been created when using `present` 

ie: phase out `modify` and merge it with `present`

Just including your formatted yaml below for easier viewing:
  

>   ```yaml
>   vars:
>     user_name: "user2"
>     group_list: "system"
>     uid: "1012"
>   
>   tasks:
>     # Add/Update users
>     - name: Add/Update user {{ user_name }}
>       ibm.power_aix.user:
>         name: "{{ user_name }}"
>         state: present
>         attributes:
>           id: "{{ uid }}"
>           groups: "{{ group_list }}"
>   ```
>   and creates the new user OK.
>   
>   However however we try to define the password it fails:
>   ```yaml
>   vars:
>     user_name: "user2"
>     group_list: "system"
>     uid: "1012"
>     password: "abc1234"
>   
>   tasks:
>     # Add/Update users
>     - name: Add/Update user {{ user_name }}
>       ibm.power_aix.user:
>         name: "{{ user_name }}"
>         state: present
>         password: "{{ password }}"
>       attributes:
>         id: "{{ uid }}"
>         groups: "{{ group_list }}"
>   ```

**Comment by cunlifs** [comment]
This is a new user, so you can't set the password at user creation (present), you have to create user the change (modify) the password?

**Comment by d-little** [comment]
Is it failing to set the password at user creation?  That should work as intended.  Are you getting an error when trying to create a new user using:

```yaml
vars:
  user_name: "user2"
  group_list: "system"
  uid: "1012"
  password: "abc1234"

tasks:
  # Add/Update users
  - name: Add/Update user {{ user_name }}
    ibm.power_aix.user:
      name: "{{ user_name }}"
      state: present
      attributes:
        id: "{{ uid }}"
        groups: "{{ group_list }}"
        password: "{{ password }}"

```

**Comment by cunlifs** [comment]
Password isn't a sub-attribute value, its a main attribute:

# ansible-doc ibm.power_aix.user -s
- name: Create new users or change/remove attributes of users on AIX.
  user:
      attributes:        # Specifies the attributes to be changed or created for the user. For details on valid user attributes,
                               please refers to IBM documentation at https://www.ibm.com/support/knowle
                               dgecenter/ssw_aix_72/c_commands/chuser.html. If you have the proper
                               authority, you can set the following usual user attributes
                               account_locked, admin, admgroups, capabilities, cpu, daemon, data,
                               default_roles, dictionlist, domains, expires, fsize, fsize_hard, gecos,
                               groups, histexpire, home, id, login, loginretries, logintimes, maxages,
                               maxexpired, maxrepeats, maxulogs, minage, minalpha, mindiff, minlen,
                               minother, nofiles, nproc, pgrp, projects, pwdchecks, pwdwarntime, rcmds,
                               rlogin, roles, rss, shell, stack, su, sugroups, sysenv, threads, tpath,
                               ttys, umask, usrenv, etc.

      password:    # Specifies the encrypted string for the password to create or change the password. Can be used when
                               `state=present'.

**Comment by JLRebello** [comment]
The user module is designed to create a user and change a password in two steps, for example:
```
  vars:
    user_name: "user2"
    group_list: "system"
    uid: "1012"
    password: "abc"

  tasks:
    # Add new user
    - name: Add user {{ user_name }}
      ibm.power_aix.user:
        name: "{{ user_name }}"
        state: present
        attributes:
          id: "{{ uid }}"
          groups: "{{ group_list }}"

    # update password for new user
    - name: Update user {{ user_name }}
      ibm.power_aix.user:
        name: "{{ user_name }}"
        state: modify
        password: "{{ password }}" 
```

The module does not support creating a new user and setting the password in one step.

**Comment by pvtorres** [comment]
This problem is addressed in the pull request #171
